src/rust/plugin-registry/src/static_files/plugin.nomad

```diff
@@ -0,0 +1,180 @@
+variable "container_registry" {
+  type        = string
+  default     = "localhost:5000"
+  description = "The container registry in which we can find Grapl services."
+}
+
+variable "plugin_id" {
+  type        = string
+  description = "The ID for this plugin."
+}
+
+variable "tenant_id" {
+  type        = string
+  description = "The tenant's ID. Used in the plugin-execution-sidecar."
+}
+
+variable "plugin_artifact_url" {
+  type        = string
+  description = "The url that specifies which binary to run as the plugin."
+}
+
+variable "plugin_count" {
+  type        = number
+  default     = 1
+  description = "The number of instances of the plugin to run."
+}
+
+variable "aws_account_id" {
+  type        = string
+  description = "The account ID of the aws account that holds onto the plugin binaries."
+}
+
+locals {
+  shared_key = "grapl_secret-${uuidv4()}"
+}
+
+job "grapl-plugin" {
+  datacenters = ["dc1"]
+  namespace   = "plugin-${var.plugin_id}"
+  type        = "service"
+
+  # We'll want to make sure we have the opposite constraint on other services
+  # This is set in the Nomad agent's `client` stanza:
+  # https://www.nomadproject.io/docs/configuration/client#meta
+  constraint {
+    attribute = "${meta.is_grapl_plugin_host}"
+    value     = true
+  }
+
+  group "plugin" {
+    network {
+      port "plugin_grpc_receiver" {}
+    }
+
+    count = var.plugin_count
+
+    task "tenant-plugin-execution-sidecar" {
+      driver = "docker"
+
+      template {
+        data        = <<EOH
+      {{ $plugin_id := env "PLUGIN_ID" }}
+      {{ with secret "pki/issue/plugin_service_ca" (printf "common_name=%s.plugins.grapl.com" $plugin_id) "format=pem" }}
+        {{ .Data.certificate }}
+        {{ .Data.issuing_ca }}
+        {{ .Data.private_key }}
+      {{ end }}
+      EOH
+        destination = "${NOMAD_SECRETS_DIR}/bundle.pem"
+        change_mode = "restart"
+      }
+
+      config {
+        image = "grapl/plugin-execution-sidecar"
+        ports = [
+        "plugin_sidecar_grpc_receiver"]
+      }
+
+      env {
+        TENANT_ID     = "${var.tenant_id}"
+        PLUGIN_ID     = "${var.plugin_id}"
+        BOOTSTRAP_KEY = "${local.shared_key}"
+      }
+    }
+
+
+    task "tenant-plugin-bootstrap-sidecar" {
+      driver = "docker"
+
+      template {
+        data        = <<EOH
+      {{ $plugin_id := env "PLUGIN_ID" }}
+      {{ with secret "pki/issue/plugin_service_ca" (printf "common_name=%s.plugins.grapl.com" $plugin_id) "format=pem" }}
```

Comment:
Based on this it looks like this might issue the same certificate for the execution sidecar and the bootstrap sidecar, right? We should issue two separate certificates I think, even if they're both for the same common name.

Reply:
So like
pki/issue/plugin_service_execution_ca
pki/issue/plugin_service_bootstrap_ca
?